packages/11ty/_plugins/iiif/manifest/index.js

```diff
@@ -0,0 +1,162 @@
+      }
+      const label = this.getAnnotationLabel(item)
+      const { name } = path.parse(src)
+      const choiceId = new URL([this.iiifConfig.inputDir, src].join('/'), process.env.URL).href
+      const format = mime.lookup(src)
+      const choice = {
+        id: choiceId,
+        format,
+        height: this.canvas.height,
+        type: 'Image',
+        label: { en: [label] },
+        width: this.canvas.width
+      }
+      if (this.figure.preset === 'zoom') {
+        const serviceId = new URL([this.iiifConfig.outputDir, name, this.iiifConfig.imageServiceDirectory].join('/'), process.env.URL)
+        choice.service = [
+          {
+            id: serviceId,
+            type: 'ImageService3',
+            profile: 'level0'
+          }
+        ]
+      }
+      return choice
+    })
+
+    return this.createAnnotation({
+      body: {
+        items,
+        type: 'Choice',
+      },
+      id: 'choices',
+      motivation: 'painting',
+    })
+  }
+
+  async create() {
+    const { height, width } = await this.getCanvasMetadata()
+    const manifest = builder.createManifest(this.manifestId, (manifest) => {
+      manifest.addLabel(this.figure.label, this.iiifConfig.locale)
+      manifest.createCanvas(this.canvas.id, (canvas) => {
+        canvas.height = height
+        canvas.width = width
+        if (this.annotations) {
+          this.annotations.forEach((item) => {
+            canvas.createAnnotation(item.id, item)
+          })
+        }
+        if (this.choices) {
+          canvas.createAnnotation(this.choices.id, this.choices)
+        }
+      })
+    })
+    this.manifest = manifest
+  }
+
+  createAnnotation({ body, id, motivation }) {
+    return {
+      body: body,
+      id: [this.canvas.id, id].join('/'),
+      motivation: motivation,
+      type: 'Annotation'
+    }
+  }
+
+  getAnnotationLabel({ label, src }) {
+    const filename = src ? path.parse(src).name : null
+    return label || titleCase(filename)
+  }
+
+  /**
+   * Use dimensions of figure.src or first choice as canvas dimensions
+   */
+  get canvasImagePath() {
+    const firstChoice = this.figure.annotations
+      .flatMap(({ items }) => items)
+      .find(({ target }) => !target)
+    const imagePath = this.figure.src || firstChoice.src
+    if (!imagePath) {
+      error(`Invalid figure ID "${this.figure.id}". Figures with annotations must have "choice" annotations or a "src" property.`)
+    }
+    return imagePath
+  }
+
+  async getCanvasMetadata() {
+    const { inputDir, inputRoot } = this.iiifConfig
+    const fullImagePath = path.join(inputRoot, inputDir, this.canvasImagePath)
+    const { height, width } = await sharp(fullImagePath).metadata()
+    this.canvas.height = height
+    this.canvas.width = width
+    return { height, width }
+  }
+
+  async toJSON() {
+    await this.create()
+    return builder.toPresentation3(this.manifest)
+  }
```

Comment:
Renaming the `create` method `toJSON` and moving the `return builder.toPresentation3(this.manifest)` statement there would make the purpose of the class more clear.